src/views/demo-page/demo-page.js

```diff
@@ -7,16 +7,72 @@ import '../../components/diff/ai-diff';
 import '../../components/select/ai-select';
 import '../../components/tabview/ai-tabview';
 
-import { getChangesModel, select } from '../../enigma/models';
-
-// apply selection composition Sense
-select('Composition', 'Sense');
+import { getChangesModel } from '../../enigma/models';
 
 const tabs = [{ title: 'Purple', content: '#A577EF' },
   { title: 'Mint', content: '#5BBA92' },
   { title: 'Forest', content: '#90BF4E' }];
 
 customElements.define('view-demo-page', class extends AIHTMLElement {
+  constructor() {
+    super();
+
+    this.table = null;
+  }
+
+  connectedCallback() {
+    this.getModel().then((model) => {
+      this.getTable(model).then((table) => {
+        this.table = table;
+        this.invalidate();
+      });
+    });
+    this.invalidate();
+  }
+
+  getModel() {
+    return getChangesModel().then((model) => {
+      model.on('changed', () => {
+        this.getTable(model).then((table) => {
+          this.table = table;
+          this.invalidate();
+        });
+      });
+      return model;
+    });
+  }
+
+  getTable(model) {
+    return model.getLayout().then((layout) => {
+      const data = layout.qHyperCube.qPivotDataPages[0]
+        ? layout.qHyperCube.qPivotDataPages[0] : null;
+      if (data) {
+        const tblHeaders = new Set(data.qTop.map(el => (el.qText || '')));
```

Comment:
Do we gain any benefits from `Set` "muscles" when we just store normal text from el.qText in our array?

does not a regular map works? 
```suggestion
        const tblHeaders = data.qTop.map(el => (el.qText || ''));
````

Reply:
not my code but I think it looks better with map. @aihazm 

Aslo, this is the demo page so it wont matter shortly. Maybe we should remove it soon?

Reply:
We usually use new Set() in order to easily create arrays with unique values, in this case, shouldn't contain duplicate values ...anyway  I agree with @snurrfint demo page will go away